caravel/assets/javascripts/explorev2/components/ChartControl.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import Select from 'react-select';
+import { bindActionCreators } from 'redux';
+import * as actions from '../actions/exploreActions';
+import { connect } from 'react-redux';
+import { VIZ_TYPES } from '../../../utils/common';
+
+class ChartControl extends React.Component {
+  componentWillMount() {
+    if (this.props.datasourceId) {
+      this.props.actions.setFormOpts(this.props.datasourceId, this.props.datasourceType);
```

Comment:
This didn't work.. Suggested by the redux doc here: [http://redux.js.org/docs/basics/Actions.html#action-creators](url)
We should bind action creators to dispatch() function, then map dispatch to component's props. I could still explicitly import actions like:
```import { setFormOpts, setDatasource, resetFormData, setVizType } from '../actions/exploreActions';```

and then do this in mapDispatchToProps: 
```actions: bindActionCreators({
      setFormOpts,
      setDatasource,
      resetFormData,
      setVizType,
    }, dispatch),```
But dispatching actions would still be the same: 
```this.props.actions.setDatasource(val);```

Do you think I should proceed with this? @ascott